From d74fb43d8cd55ea470441013a533d4108a2128e4 Mon Sep 17 00:00:00 2001
From: Olaf Hering <olaf@aepfle.de>
Date: Mon, 30 Nov 2020 20:12:36 +0100
Subject: [PATCH] Remove GSourceFunc casts

Some function prototypes are already a GSourceFunc, no need to cast the pointer.

Adjust other places to convert the callback into a real GSourceFunc.

Signed-off-by: Olaf Hering <olaf@aepfle.de>
---
 panel-plugin/weather-config.c  |  5 ++---
 panel-plugin/weather-summary.c | 10 ++++++++--
 panel-plugin/weather.c         |  8 ++++----
 3 files changed, 14 insertions(+), 9 deletions(-)

diff --git a/panel-plugin/weather-config.c b/panel-plugin/weather-config.c
index 67db5095..31ef7e1e 100644
--- a/panel-plugin/weather-config.c
+++ b/panel-plugin/weather-config.c
@@ -150,7 +150,7 @@ update_summary_window(xfceweather_dialog *dialog,
 static gboolean
 schedule_data_update(gpointer user_data)
 {
-    xfceweather_dialog *dialog = (xfceweather_dialog *) user_data;
+    xfceweather_dialog *dialog = user_data;
     plugin_data *pd = dialog->pd;
 
     /* force update of downloaded data */
@@ -192,8 +192,7 @@ schedule_delayed_data_update(xfceweather_dialog *dialog)
     gtk_widget_show(GTK_WIDGET(dialog->update_spinner));
     gtk_spinner_start(GTK_SPINNER(dialog->update_spinner));
     dialog->timer_id =
-        g_timeout_add_seconds(UPDATE_TIMER_DELAY,
-                              (GSourceFunc) schedule_data_update, dialog);
+        g_timeout_add_seconds(UPDATE_TIMER_DELAY, schedule_data_update, dialog);
 }
 
 
diff --git a/panel-plugin/weather-summary.c b/panel-plugin/weather-summary.c
index 2ceef322..aa428f1e 100644
--- a/panel-plugin/weather-summary.c
+++ b/panel-plugin/weather-summary.c
@@ -1120,6 +1120,13 @@ cb_notebook_page_switched(GtkNotebook *notebook,
 }
 
 
+static gboolean
+update_summary_subtitle_cb(gpointer user_data)
+{
+    plugin_data *data = user_data;
+    return update_summary_subtitle(data);
+}
+
 gboolean
 update_summary_subtitle(plugin_data *data)
 {
@@ -1154,8 +1161,7 @@ update_summary_subtitle(plugin_data *data)
 #endif
         update_interval = 1000 - (now_ms % 1000) + 10;
     data->summary_update_timer =
-        g_timeout_add(update_interval, (GSourceFunc) update_summary_subtitle,
-                      data);
+        g_timeout_add(update_interval, update_summary_subtitle_cb, data);
     return FALSE;
 }
 
diff --git a/panel-plugin/weather.c b/panel-plugin/weather.c
index 35182527..096e0204 100644
--- a/panel-plugin/weather.c
+++ b/panel-plugin/weather.c
@@ -586,8 +586,9 @@ cb_weather_update(SoupSession *session,
 
 
 static gboolean
-update_handler(plugin_data *data)
+update_handler(gpointer user_data)
 {
+    plugin_data *data = user_data;
     gchar *api_version = FORECAST_API;
     gchar *url;
     gboolean night_time;
@@ -759,8 +760,7 @@ schedule_next_wakeup(plugin_data *data)
 
     date = format_date(now_t, "%Y-%m-%d %H:%M:%S", TRUE);
     data->update_timer =
-        g_timeout_add_seconds((guint) diff,
-                              (GSourceFunc) update_handler, data);
+        g_timeout_add_seconds((guint) diff, update_handler, data);
     if (!strcmp(data->next_wakeup_reason, "regular check"))
         weather_debug("[%s]: Running regular check for updates, "
                       "interval %d secs.", date, UPDATE_INTERVAL);
@@ -1513,7 +1513,7 @@ void
 forecast_click(GtkWidget *widget,
                gpointer user_data)
 {
-    plugin_data *data = (plugin_data *) user_data;
+    plugin_data *data = user_data;
 
     if (data->summary_window != NULL)
         gtk_widget_destroy(data->summary_window);
-- 
GitLab


